subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -1146,11 +1140,9 @@ uint8_t ll_adv_aux_sr_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref,
 			len = ad_len_chain;
 		}
 
-		/* PDU length so far */
-		sec_len = dptr_chain - &pdu_chain->payload[0];
-
 		/* Check AdvData overflow */
-		if ((sec_len + len) > PDU_AC_PAYLOAD_SIZE_MAX) {
+		if ((sec_len + ad_len_overflow + len) >
+		    PDU_AC_PAYLOAD_SIZE_MAX) {
```

Comment:
Any risk of overflow here when comparing?

Reply:
`sec_len` is `uint16_t`; sum of `sec_len` (<256) , `ad_len_overflow` (<256) and `len` (<256) does not exceed `UINT16_MAX`